test/shopify-cli/js_system_test.rb

```diff
@@ -76,8 +76,39 @@ def test_yarn_on_class_proxies_to_the_instance_version_of_call
       JsSystem.yarn?(@context)
     end
 
+    def test_yarn_check_returns_false_if_yarn_lock_not_present
+      mock_yarn_check(lock_exists: false, status: stubs(success?: true))
+
+      refute JsSystem.yarn?(@context)
+    end
+
+    def test_yarn_check_returns_false_if_yarn_lock_present_but_which_yarn_call_fails
```

Comment:
combinatorially, we are missing one test here. The "yarn check returns false if yarn lock not present" case presupposes knowledge of the implementation, knowing that it short circuits the call.  We need to check that case where both are false, and the case where only the yarn lock is missing

logically:

true true
true false
false true
false false

we only test three right now

Reply:
Added the final conditional check.